### PR TITLE
#1367 change log4j-1.2.17 to log4j-2.17.1

### LINKED ISSUE #1367: Log4j Critical Exploit - Play1 possibly unaffected
A Log4j critical exploit was made public, here's a good overview: https://www.lunasec.io/docs/blog/log4j-zero-day/ I believe the minimum version number where the exploit is functional is 2.0 and I believe play1 is on 1.2.17, so it _shouldn't_ be a problem. Due to the severity of the issue I thought it best to post here and get another opinion just in case.

### PR DESCRIPTION
**Pull request by jacol84**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you updated the documentation?
* [x] Have you added tests for any changed functionality?

# Helpful things
[Migrating from Log4j 1.x to 2.x](https://logging.apache.org/log4j/2.x/manual/migration.html)

## Fixes

Fixes #1367

## Purpose

proposal to fix bugs and vulnerabilities in the code
CVE-2021-4104
CVE-2019-17571

## Background Context

a good step on update of libraries

TODO: discover if there is another path  _-Dlog4j2.configurationFile=$path$_ or if there is any of: _log4j2.xml, log4j2.properties, json, yaml, ...._ files in resources than  log4j will be automatically configured, so Play can simply use this configuration.  [link](https://logging.apache.org/log4j/2.x/manual/configuration.html)

## References
[log4j2](https://logging.apache.org/log4j/2.x/)

### CODE REVIEW

**Review by asolntsev — APPROVED** [review]
Technically, the PR is good.
But are you sure log4j2 is better than alternatives? I know many people prefer "Logback" nowadays.

**Comment by jacol84** [comment]
> may need to give users a choice
use SL4J and add add dependecy.yaml

**Comment by xabolcs** [comment]
> But are you sure log4j2 is better than alternatives? I know many people prefer "Logback" nowadays.

+1 for [Logback](https://logback.qos.ch/)!

> ### Logback Project
>
> Logback is intended as a successor to the popular log4j project, [picking up where log4j 1.x leaves off](https://logback.qos.ch/reasonsToSwitch.html).
> Logback's architecture is quite generic so as to apply under different circumstances. At present time, logback is divided into three modules, `logback-core`, `logback-classic` and `logback-access`.
>
> The `logback-core` module lays the groundwork for the other two modules. The `logback-classic` module can be assimilated to a significantly improved version of log4j 1.x. Moreover, `logback-classic` natively implements the [SLF4J API](http://www.slf4j.org/) so that you can readily switch back and forth between logback and other logging frameworks such as log4j 1.x or `java.util.logging` (JUL).
>
> The `logback-access` module integrates with Servlet containers, such as Tomcat and Jetty, to provide HTTP-access log functionality. Note that you could easily build your own module on top of `logback-core`.

**Comment by jacol84** [comment]
add link to _Migrating from Log4j 1.x to 2.x_

**Comment by jacol84** [comment]
I think "PLAY" was using log4j so this is upgrading the library to the latest version - better for sure than staying in the old version
@asolntsev 
and the best solution would be to choose which logger you want to use
if you want to use log4j2 or logback your selections should add deps

File: framework/test-src/play/testlog4j.properties [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-
-log4j.appender.PropertiesConsoleAppender=org.apache.log4j.ConsoleAppender
-log4j.appender.PropertiesConsoleAppender.layout=org.apache.log4j.PatternLayout
-log4j.appender.PropertiesConsoleAppender.layout.ConversionPattern=%d{ISO8601} %-5p [%t] %c: %m%n
-
-
-
-log4j.logger.logtest.properties=ERROR
+# appender Console
+appender.console.type = Console
```
Comment (xael-fry) [inline framework/test-src/play/testlog4j.properties:2]: Could you replace CRLF by LF ?

**Review by xael-fry — CHANGES_REQUESTED** [review]

File: framework/test-src/play/testlog4j.properties (hunk 1 above)
Comment (xael-fry) [reply framework/test-src/play/testlog4j.properties:2]: done in https://github.com/playframework/play1/pull/1380

**Comment by xael-fry** [comment]
fix conflict in https://github.com/playframework/play1/pull/1380

File: framework/test-src/play/testlog4j.properties (hunk 1 above)
Comment (jacol84) [reply framework/test-src/play/testlog4j.properties]: i changed in a64ab7c417a67c074e201f842097722791cff88f
and add solution --> when configuration not working use autoconfiguration

**Comment by xael-fry** [comment]
@jacol84 xan you rebase it to master to avoid conflict? Thanks

**Comment by jacol84** [comment]
I did a rebase
dw @xael-fry 
now has wrninng netty-3.10.6.Final.ja
`WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jboss.netty.util.internal.ByteBufferUtil (file:/S:/praca/play/git/play1MY/framework/lib/netty-3.10.6.Final.jar) to method java.nio.DirectByteBuffer.cleaner()
WARNING: Please consider reporting this to the maintainers of org.jboss.netty.util.internal.ByteBufferUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2022-02-02 09:49:21,946 [play-thread-1] INFO play - Application 'xxx' is now started !`

**Comment by jacol84** [comment]
@xael-fry 
I change  "github workflow" to correctly with documentation -- > change folder how run ant 
I fixed the test i-am-a-developer (last time I missed it)

**Review by xael-fry — APPROVED** [review]

**Pull request merged**